egs/tedlium/s5/local/nnet3/run_ivector_common.sh

```diff
@@ -7,62 +7,56 @@
-    mfccdir=mfcc_perturbed
-    steps/make_mfcc.sh --cmd "$train_cmd" --nj 50 \
-      data/${datadir}_tmp exp/make_mfcc/${datadir}_tmp $mfccdir || exit 1;
-    steps/compute_cmvn_stats.sh data/${datadir}_tmp exp/make_mfcc/${datadir}_tmp $mfccdir || exit1;
-    utils/fix_data_dir.sh data/${datadir}_tmp
+if [ $stage -le 1 ]; then
+  #Although the nnet will be trained by high resolution data, we still have to perturb the normal data to get the alignment
+
+  utils/perturb_data_dir_speed.sh 0.9 data/${train_set} data/temp1
+  utils/perturb_data_dir_speed.sh 1.1 data/${train_set} data/temp2
+  utils/combine_data.sh data/${train_set}_tmp data/temp1 data/temp2
+  utils/validate_data_dir.sh --no-feats data/${train_set}_tmp
+  rm -r data/temp1 data/temp2
+
+  mfccdir=mfcc_perturbed
+  steps/make_mfcc.sh --cmd "$train_cmd" --nj 50 \
+    data/${train_set}_tmp exp/make_mfcc/${train_set}_tmp $mfccdir || exit 1;
+  steps/compute_cmvn_stats.sh data/${train_set}_tmp exp/make_mfcc/${train_set}_tmp $mfccdir || exit1;
+  utils/fix_data_dir.sh data/${train_set}_tmp
     
-    utils/copy_data_dir.sh --spk-prefix sp1.0- --utt-prefix sp1.0- data/${datadir} data/temp0
-    utils/combine_data.sh data/${datadir}_sp data/${datadir}_tmp data/temp0
-    utils/fix_data_dir.sh data/${datadir}_sp
-    rm -r data/temp0 data/${datadir}_tmp
-  done
-  fi
+  utils/copy_data_dir.sh --spk-prefix sp1.0- --utt-prefix sp1.0- data/${train_set} data/temp0
+  utils/combine_data.sh data/${train_set}_sp data/${train_set}_tmp data/temp0
+  utils/fix_data_dir.sh data/${train_set}_sp
+  rm -r data/temp0 data/${train_set}_tmp
+fi
 
-  if [ $stage -le 2 ] && [ "$generate_alignments" == "true" ]; then
-    # obtain the alignment of the pertubed data
-    steps/align_fmllr.sh --nj 100 --cmd "$train_cmd" \
-      data/${train_set}_sp data/lang_nosp exp/tri3 exp/tri3_ali_sp || exit 1
-  fi
+train_set_sp=${train_set}_sp
 
-  train_set=${train_set}_sp
+if [ $stage -le 2 ] && [ "$generate_alignments" == "true" ]; then
+  # obtain the alignment of the pertubed data
+  steps/align_fmllr.sh --nj 100 --cmd "$train_cmd" \
+    data/${train_set_sp} data/lang_nosp exp/tri3 exp/tri3_ali_sp || exit 1
 fi
 
 if [ $stage -le 3 ]; then
+  
   mfccdir=mfcc_hires
   if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d $mfccdir/storage ]; then
     date=$(date +'%m_%d_%H_%M')
-    utils/create_split_dir.pl /export/b{09,10,11,12}/$USER/kaldi-data/egs/tedlium-$date/s5b/$mfccdir/storage $mfccdir/storage
+    utils/create_split_dir.pl /export/b{09,10,11,12}/$USER/kaldi-data/egs/tedlium-$date/s5/$mfccdir/storage $mfccdir/storage
   fi
 
-  for dataset in $train_set dev test; do
-    utils/copy_data_dir.sh data/$dataset data/${dataset}_hires
-
+  for dataset in $train_set $train_set_sp; do
     data_dir=data/${dataset}_hires
+    utils/copy_data_dir.sh data/$dataset $data_dir
+
       # this next section does volume perturbation on the data.
     cat $data_dir/wav.scp | python -c "
 import sys, os, subprocess, re, random
```

Comment:
You have to pass a seed to the random number generator to ensure reproducibility. Use random.seed(0)